dataproc/single_job_workflow.py

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/env python
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""Sample Cloud Dataproc inline workflow to run a pyspark job on an ephermeral
+cluster.
+
+Example Usage to run the inline workflow on a managed cluster:
+python single_job_workflow.py --project_id=$PROJECT --gcs_bucket=$BUCKET \
+  --cluster_name=$CLUSTER --zone=$ZONE
+
+Example Usage to run the inline workflow on a global region managed cluster:
+python submit_job_to_cluster.py --project_id=$PROJECT --gcs_bucket=$BUCKET \
+  --cluster_name=$CLUSTER --zone=$ZONE --global_region
+
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+import argparse
+import os
+from google.cloud import dataproc_v1
+from google.cloud import storage
+from google.cloud.dataproc_v1.gapic.transports import (
+    workflow_template_service_grpc_transport,
+)
+
+DEFAULT_FILENAME = "pyspark_sort.py"
+waiting_callback = False
+
+
+def get_default_pyspark_file():
+    """Gets the PySpark file from this directory."""
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    f = open(os.path.join(current_dir, DEFAULT_FILENAME), "rb")
+    return f, DEFAULT_FILENAME
+
+
+def get_pyspark_file(filename):
```

Comment:
Let's refactor this a little bit so we don't need to pick which function (get_pyspark_file or get_default_pyspark_file) we have to call. Then, the first part of the body of this function becomes:

```
if not filename:
    current_dir = os.path.dirname(os.path.abspath(__file__))
    filename = os.path.join(current_dir, DEFAULT_FILENAME)
```

Reply:
Did this is a similar manner to your suggestion in response to previous comment.